tests/server/init.js

```diff
@@ -0,0 +1,31 @@
+import fs from 'fs';
+import path from 'path';
+
+
+/*
+ * Throw an error if there are local configs that would pollute the test suite.
+ */
+export function checkLocalConfigs() {
+  const root = path.resolve(path.join(path.dirname(__filename), '..', '..'));
+  if (!fs.statSync(root).isDirectory()) {
+    throw new Error(`Oops, detected the wrong root? ${root}`);
+  }
+
+  const configDir = path.join(root, 'config');
+  if (!fs.statSync(configDir).isDirectory()) {
+    throw new Error(`Oops, detected the wrong config dir? ${configDir}`);
+  }
+  const disallowedFiles = fs.readdirSync(configDir)
+    // Check for local configs but allow dev configs.
+    .filter((name) => name.startsWith('local-') && !name.startsWith('local-dev'))
```

Comment:
Maybe this should be `name.startsWith('local')`? Otherwise a `local.json` would get loaded.